Tasks/Gradle/gradletask.ts

```diff
@@ -4,88 +4,165 @@ import tl = require('vsts-task-lib/task');
+    }
 }
 
-var publishJUnitResults = tl.getInput('publishJUnitResults');
-var testResultsFiles = tl.getInput('testResultsFiles', true);
-
-function publishTestResults(publishJUnitResults, testResultsFiles: string) {
-  if(publishJUnitResults == 'true') {
-    //check for pattern in testResultsFiles
-    if(testResultsFiles.indexOf('*') >= 0 || testResultsFiles.indexOf('?') >= 0) {
-      tl.debug('Pattern found in testResultsFiles parameter');
-      var buildFolder = tl.getVariable('agent.buildDirectory');
-      var allFiles = tl.find(buildFolder);
-      var matchingTestResultsFiles = tl.match(allFiles, testResultsFiles, { matchBase: true });
+function extractJavaHome() {
+    if (javaHomeSelection == 'JDKVersion') {
+        tl.debug('Using JDK version to find and set JAVA_HOME');
+
+        if (jdkVersion != 'default') {
+            // jdkVersion should be in the form of 1.7, 1.8, or 1.10
+            // jdkArchitecture is either x64 or x86
+            // envName for version 1.7 and x64 would be "JAVA_HOME_7_X64"
+            var envName = "JAVA_HOME_" + jdkVersion.slice(2) + "_" + jdkArchitecture.toUpperCase();
+            specifiedJavaHome = tl.getVariable(envName);
+            if (!specifiedJavaHome) {
+                tl.error('Failed to find specified JDK version. Please make sure environment variable ' + envName + ' exists and is set to the location of a corresponding JDK.');
+                tl.exit(1);
+            }
+        }
     }
     else {
-      tl.debug('No pattern found in testResultsFiles parameter');
-      var matchingTestResultsFiles = [testResultsFiles];
+        tl.debug('Using path from user input to set JAVA_HOME');
+        var jdkUserInputPath = tl.getPathInput('jdkUserInputPath', true, true);
+        specifiedJavaHome = jdkUserInputPath;
     }
 
-    if(!matchingTestResultsFiles) {
-      tl.warning('No test result files matching ' + testResultsFiles + ' were found, so publishing JUnit test results is being skipped.');  
-      return 0;
+    return specifiedJavaHome;
+}
+
+function enableCodeCoverage() {
+    var reportDir = path.join(cwd + "/" + uuid.v1());
+    fs.mkdirSync(reportDir);
+
+    var isMultiModule = isMultiModuleProject(wrapperScript);
+
+    if (ccTool.toLowerCase() == "jacoco") {
+        if (isMultiModule) {
+            ccReportingTask = "jacocoRootReport";
+        } else {
+            ccReportingTask = "jacocoTestReport";
+        }
+        summaryFile = path.join(reportDir + "/summary.xml");
+    } else {
+        summaryFile = path.join(reportDir + "/coverage.xml")
+        ccReportingTask = "cobertura"
     }
 
-    var tp = new tl.TestPublisher("JUnit");
-    tp.publish(matchingTestResultsFiles, false, "", "", "", "");
-  } 
+    var buildProps: { [key: string]: string } = {};
+    buildProps['classfiledirs'] = classFileDirs;
+    buildProps['classfilters'] = classFilters;
+    buildProps['summaryfile'] = summaryFile;
+    buildProps['reportdir'] = reportDir;
```

Comment:
cleaner way is to add all these varables classFileDirs, classFilters, summaryFile, reportDir, isMultimodule to a class and pass the class to this method.